File: seminar/src/main/java/com/server/dosopt/seminar/service/CustomerService.java
package com.server.dosopt.seminar.service;

import com.server.dosopt.seminar.domain.CustomerEntity;
import com.server.dosopt.seminar.dto.request.customer.CustomerRequest;
import com.server.dosopt.seminar.repository.CustomerJpaRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Slf4j
@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class CustomerService {

    private final CustomerJpaRepository customerJpaRepository;

    @Transactional
    public void create(CustomerRequest request) {
        CustomerEntity customer = CustomerEntity.builder()
                .name(request.name())
                .nickname(request.nickname())
                .age(request.age())
                .build();
        log.info("customer: {}", customer);
        customerJpaRepository.save(customer);
    }
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/service/CustomerService.kt
package sopt.org.kotlinseminar.service

import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import sopt.org.kotlinseminar.domain.CustomerEntity
import sopt.org.kotlinseminar.dto.request.customer.CustomerRequest
import sopt.org.kotlinseminar.repository.CustomerJpaRepository

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
class CustomerService(
        private val customerJpaRepository: CustomerJpaRepository
) {

    @Transactional
    fun create(request: CustomerRequest) {
        val customer = CustomerEntity(request.name, request.age, request.nickname)
        println(customer)
        customerJpaRepository.save(customer)
    }
}